docs/developer/architecture/user_customisations.rst

```diff
@@ -0,0 +1,51 @@
+User Customisations and Persistence
+===================================
+
+Context
+-------
+
+MetalK8s comes with a set of tools and services that may need to be configured
+on site. At the same time, we don't want the administrator of the cluster to
+master each and every service of the cluster. We also don't want to allow any
+kind of customisations since it would make the system even more complex to test
+and maintain over the time.
+
+In addition to those services, MetalK8s deployment may have to be adapted
```

Comment:
I'd like more details about what you call "MetalK8s configurations"... Something else than what we have in BootstrapConfig + Node objects?

Reply:
So I was mainly referring to the number of replicas for the MetalK8s services with deploy such as Prometheus or AlertManager. It is true that content of BootstrapConfig or NodeObjects config (roles and taints) as well as solution configs (what solution is available on the bootstrap or envs or namespaces created when deploying solutions) are also part of MetalK8s settings we need to expose to the MetalK8s administrator and that need to persist. Will update this part to be more precise.

Reply:
done